### ISSUE TITLE
XSS Vulnerability

### ISSUE THREAD

**Issue by licatajustin** [ifad/data-confirm-modal#72]
If the contents of the `data-confirm` attribute contain user-generated content, this library opens up the app to possible XSS vulnerabilities, which makes sense given [the use of the `html()` function](https://github.com/ifad/data-confirm-modal/blob/52b12473bb110e2437b06e262b0435741ceac75c/vendor/assets/javascripts/data-confirm-modal.js#L212) here.

Is it by design to allow `script` tags to execute if passed into the body? I understand the desire to pass in HTML that is rendered properly in the modal body, but perhaps we can work to prevent these `script` tags from also executing. Thoughts?

Crude Example:

```erb
<% # Assumption: user.name = "<script>alert('oops')</script>" %>

<%= link_to("Open", "#", data: { confirm: "Are you sure you want to open this #{user.name}?" }) %>
```

**Comment by vjt** [comment]
Hi, many thanks for the heads up. I reviewed the approach in the forked repo, and as we all know [you can't parse [X|HT]ML with regexen](https://stackoverflow.com/a/1732454/69379).

There are some interesting solutions [here](https://stackoverflow.com/questions/6659351/removing-all-script-tags-from-html-with-js-regular-expression) and [here](https://gist.github.com/sindresorhus/1993156) and [here](https://api.jquery.com/jquery.parsehtml/) but I still haven't made up my mind on this, because there's `onload`, `onerror`, etc.

Any solution has its drawbacks, as this is XML and it is complex. Maybe the best solution would be a `README` entry saying that the body of the modal must not be user-controlled as this will lead to XSS. The real solution is to embed a full HTML sanitiser, but again I don't like that because of the corner case of having the end user provide the modal body.

**Comment by licatajustin** [comment]
@vjt, fair point -- `onerror` is certainly still an issue with the temporary patch I applied in the forked repo. What are your thoughts on adding a dependency like [DOMPurify](https://github.com/cure53/DOMPurify)?
